### PR TITLE
Add the JetBrains CLion IDE

### PR DESCRIPTION
**Pull request by AbigailBuccaneer**

### CODE REVIEW

**Comment by didrocks** [comment]
Hey AbigailBuccaneer,
Thanks for this pull requests and helping Ubuntu Make to be better! JetBrains CLion IDE support would be a nice additional feature!

I'll just add one comment on the commit itself in a sec, however, things looks good.
This though, is missing tests (as we download from the network, we need to ensure that the framework is always installable).

For this, we have large and medium tests.
You can find other JetBrains based ide tests in tests/large/test_ide.py and tests/medium/test_ide.py. If you follow what is done for them, this should be quite straightfoward :)

Note that medium tests need mock assets to run inside a docker container. You will find what is used for other JetBrains frameworks in tests/data/server-content/www.jetbrains.com/, just add the needed ones for CLion.

Then, to run those tests, you can execute, for instance: ./runtests tests/large/test_ide.py:<class name>.

Do not hesitate if you need any help!

File: umake/frameworks/ide.py [hunk 1]
```diff
@@ -338,6 +338,22 @@ def __init__(self, category):
...
                          icon_filename='webide.png')
 
+class CLion(BaseJetBrains):
+    """The JetBrains CLion IDE"""
+    download_page_url = "https://www.jetbrains.com/clion/download/download_thanks.jsp?os=linux"
+    executable = "clion.sh"
+
+    def __init__(self, category):
+        super().__init__(name="CLion",
+                         description=_("CLion"),
```
Comment (didrocks) [inline umake/frameworks/ide.py]: I would add a little bit more in description, like "Fully integrated C/C++ IDE" (from their website). Sounds ok?

Comment (AbigailBuccaneer) [inline umake/frameworks/ide.py]: I used this description to be consistent with the other JetBrains IDEs in this file, but I can change it to something more descriptive if you'd like.

Comment (didrocks) [inline umake/frameworks/ide.py]: Yes please, that would be useful! (I will maybe change other as well)

**Comment by AbigailBuccaneer** [comment]
These unit tests are giving me serious grief. After having done all the `virtualenv` stuff and installed all the packages it lists, I find:

* `./runtests tests/medium/test_ide.py` doesn't run if not as root. (`subprocess.CalledProcessError: Command '['/usr/bin/docker', 'run', ...`] returned non-zero exit status 1`)
* When running the medium IDE tests as root, it sits around for a long time, taking my network connection down and up (which is very infuriating when I've got work to be doing), printing a newline to the terminal occasionally and then declares that every test failed due to timing out.
* `./runtests tests/large/test_ide.py` runs, and then hangs at:

```
[umake.frameworks] DEBUG: Call run_for on Arduino
[umake.frameworks] DEBUG: Requesting root access
[sudo] password for abigail:

```

...and I can't actually enter my password.

* If I run the large IDE tests as root, it hangs at: `Eclipse is already installed on your system, do you want to reinstall it anyway? [Yes (y)/No (N)]` and then eventually dies presumably of timeout, and then 9 of the tests failed and 1 errored.

(I did not have Eclipse installed on this system prior to this; this installation of Ubuntu is less than 24 hours old.)

I've got an actual job to be doing - unless there's a clear fix for the catastrophic test outcomes I'm seeing, I can't justify spending any more time on this PR.

**Comment by didrocks** [comment]
Those tests are not unit tests but integration one. If you feel that the README.md isn't clear enough about it (see "Tests/Types of tests"), please let me know and I'll rephrase.

* You don't need to be root to run medium tests if you are on a distribution like ubuntu, where the docker group is added to your user when you install it. Of course, if you just installed docker, you need to have a new login session for this.
* Also, the "taking network down and up" isn't the way docker works: it's creating new virtual network bridges on the fly for each tests (and containers spawn up). You have a notification about it, but it doesn't tear down your real wifi/ethernet connection, and so you can continue working. :)
* I seriously encourage to only run one tests (or a category test) as I was advising above. As part of the README.md, I gave an example of this: "Tests/Running some tests with all debug infos" (so provide the class name).

-> That will help to know why the tests are timing out for you if you can run for one framework and paste the output.

* On the large tests: same, only run it for one. If you had the `Eclipse is already installed` output, I guess you installed it by running the tests once first (and interrupting it in the middle). The large tests are really installing in your system (contrary to medium tests which fake assets and servers while downloading on the container). That's why if you have packages required but not installed, it will ask you for the root (and that's why it's running as root on jenkins).

So, I would:
* focus on the large test first, try running one for another netbeans frameworks, like:
sudo ./runtests tests/large/test_ide.py:WebStormIDETests.test_default_install (you can omit sudo if all dependencies are already installed).
If that one pass, copy it and adapt to CLion
* try only then the medium test.

Ensure as well that you have pexpect >=4 (the requirements.txt should do it anyway). Those 800 tests (400 for each archs) are running every 3 hours in the jenkins datacenter, and they are all passing for now (apart from webstorm which released a new version with an icon name change that I'm fixing). So I don't expect for tests to be a disaster or anything.
If I can do anything to help you, do not hesitate to ask! Thanks again.

**Comment by didrocks** [comment]
Hey! Just coming to the news if you fixed your environment issues :)

If you don't think you will have time to work on this, I'm happy to add the large and medium tests as needed, and just merging this commit. Just keep me posted, thanks!

**Comment by AbigailBuccaneer** [comment]
I don't have the time to work on this - please feel free to take this commit and add tests as needed. Thanks / sorry!

**Comment by didrocks** [comment]
No worry! Thanks for the initial work. I'll handle this later this week then. :)

**Comment by didrocks** [comment]
Added this morning the large and medium tests (+ a pep8 fix) and it's now merged! Thanks a lot for doing the initial work :)

**Pull request merged**